### PR TITLE
Made the menu checker behave affirmatively even if it's globally unanimous

### LINKED ISSUE #5128: Dashboard GROUPS with roles are treated with "AND"
#### Sonata packages

```
sonata-project/admin-bundle              3.x-dev 82a4d6e dev-master 4b2e943 The missing Symfony Admin Generator
sonata-project/block-bundle              3.12.1          3.12.1             Symfony SonataBlockBundle
sonata-project/cache                     2.0.1           2.0.1              Cache library
sonata-project/core-bundle               3.9.1           3.10.0             Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.3.1           2.3.1              Symfony SonataDatagridBundle
sonata-project/doctrine-orm-admin-bundle 3.x-dev ec1d2a6 dev-master 1468a90 Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  1.9.0           1.9.0              Lightweight Exporter library
```

#### Symfony packages

```
Too much of it, 4.0.11 basically.
```

#### PHP version

```
PHP 7.2.6
```

## Subject

Dashboard groups with roles are treated with "AND"

## Steps to reproduce

```yaml
sonata_admin:
  dashboard:
    groups:
      app.admin.dashboard:
        label: app.admin.dashboard
        icon:  '<i class="fa fa-bar-chart-o"></i>'
        items:
          - route: sonata_admin_dashboard
            label: app.admin.dashboard
        roles: [ROLE_SYS_ADMIN, ROLE_CLIENT_ADMIN]
        on_top: true
```

## Expected results

Both ROLE_SYS_ADMIN and ROLE_CLIENT_ADMIN see the menu item.

## Actual results

Only a user with both roles sees the menu item.

Caused by https://github.com/sonata-project/SonataAdminBundle/pull/5086

### PR DESCRIPTION
**Pull request by andrewmy**
I am targeting this branch, because it's a fix.

Closes #5128

## Changelog

```markdown
### Fixed
- Menu item security was expected to be checked affirmatively rather than unanimously
```
## Subject

Menu item security was expected to be checked affirmatively rather than unanimously, fixed it.

### CODE REVIEW

**Comment by jordisala1991** [comment]
Make sure we want to change this behavior, this has nothing to do with #5086 . It was like this before this PR too.

I am +1 for this PR, but it could be considered a BC break. (before you only got access if you had all the roles, and now you have access with only one role. If someone was expecting the old behavior...)

**Comment by andrewmy** [comment]
Before #5086 you got access with one role too.

**Comment by OskarStark** [comment]
> Before #5086 you got access with one role too.

that is my understanding, too @jord
and my code results in the same with 3.25.1 and **this PR**

**Comment by andrewmy** [comment]
https://github.com/sonata-project/SonataAdminBundle/compare/3.35.1...3.35.2#diff-c81e2b98d92dc43dc33ffe99f65e8f5e — if the group was `on_top`, security didn't matter at all

**Comment by jordisala1991** [comment]
Well, that's a different issue, and clearly a bug... that's what was about my PR (unification and reusability of the code)

**Comment by andrewmy** [comment]
Ready to merge?

File: tests/Menu/Provider/GroupMenuProviderTest.php [hunk 1]
```diff
@@ -340,6 +377,49 @@ public function getAdminGroups()
...
 
+    /**
+     * @return array
+     */
+    public function getAdminGroupsMultipleRoles()
+    {
+        return [
+            [
+                'bar' => [
+                    'label' => 'foo',
```
Comment (OskarStark) [inline tests/Menu/Provider/GroupMenuProviderTest.php]: @andrewmy maybe another test with `on_top` = `true`?

Comment (andrewmy) [reply tests/Menu/Provider/GroupMenuProviderTest.php]: @OskarStark take a look

File: src/Menu/Provider/GroupMenuProvider.php [hunk 2]
```diff
@@ -134,10 +134,28 @@ private function canGenerateMenuItem(array $item, array $group)
...
-        );
+        if (null === $this->checker) {
+            return true;
+        }
+
+        // Making the checker behave affirmatively even if it's globally unanimous
+        $roles = null;
+        if (!empty($item['roles'])) {
+            $roles = $item['roles'];
+        } elseif (!empty($group['roles'])) {
```
Comment (jordisala1991) [inline src/Menu/Provider/GroupMenuProvider.php]: This means you can't have group + item roles defined...

Comment (andrewmy) [reply src/Menu/Provider/GroupMenuProvider.php]: True, yet I'm following the same flow as before the change

Comment (jordisala1991) [reply src/Menu/Provider/GroupMenuProvider.php]: Not true, before your change, both item and group roles were checked.

Comment (andrewmy) [reply src/Menu/Provider/GroupMenuProvider.php]: My bad, thanks for pointing this out. @jordisala1991 take a look again

**Comment by andrewmy** [comment]
Ready to merge?

**Comment by greg0ire** [comment]
Thanks @andrewmy !

**Pull request merged**